packages/python-google-compute-engine/google_compute_engine/instance_setup/instance_setup.py

```diff
@@ -140,6 +155,27 @@ def _GenerateSshKey(self, key_type, key_dest):
 
     file_utils.SetPermissions(key_dest, mode=0o600)
     file_utils.SetPermissions('%s.pub' % key_dest, mode=0o644)
+    with open('%s.pub' % key_dest, 'r') as pk:
+        key_data = pk.read()
+
+    key_values = key_data.split()
+    if len(key_values) < 2:
+        return
+    else:
+        return key_values[0], key_values[1]
```

Comment:
key_type is passed in; why return it?

Reply:
Sanity check. The passed in key_type is used to find the right file to read, while the value that is returned is from the contents of the file. This shouldn't ever be different, and I don't have a strong attachment to doing it this way, so if you would prefer that this only return the key value, I'm ok with that.

Reply:
i'm not opposed, just unclear why you'd return something you're passing in. maybe add a comment explaining but otherwise your explanation is good enough for me.